src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Note<T> first;
+    private Note<T> last;
+    private int size;
+    private Note<T> boofNode;
+    private Note<T> newNode;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        }
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        isIndexOutOfBounds(index);
+        Note<T> oldNote = findeIndex(index);
+        newNode = new Note<>(oldNote.prev, value, oldNote);
+        oldNote.prev.next = newNode;
+        oldNote.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T newNode : list) {
+            linkLast(newNode);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexOutOfBounds(index);
+        return findeIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexOutOfBounds(index);
+        boofNode = findeIndex(index);
+        T oldValue = boofNode.value;
+        boofNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexOutOfBounds(index);
+        boofNode = findeIndex(index);
+        T removedValue = boofNode.value;
+        if (index == 0) {
+            first = first.next;
+        } else if (index == size - 1) {
+            last = last.prev;
+        } else {
+            boofNode.prev.next = boofNode.next;
+            boofNode.next.prev = boofNode.prev;
+        }
+        size--;
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            boofNode = findeIndex(i);
+            if (object == boofNode.value || object != null && object.equals(boofNode.value)) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Note<T> findeIndex(int index) {
+        boofNode = first;
+        if (index == 0) {
+            return boofNode;
+        }
+        int count = 1;
+        while (count <= index) {
```

Comment:
Depending to the index you receive as an input, you can iterate only first or second half of the list (checklist). Maybe it can be implemented in your case as well.